server/tests/src/test/java/org/infinispan/server/functional/rest/XSiteRestMetricsOperations.java

```diff
@@ -138,36 +144,80 @@ public void testMetricExists() {
       //noinspection resource
       SERVERS.rest(NYC).withServerConfiguration(new StringConfiguration(nycXML)).create();
 
-      Map<String, String> metricsAndTags = getMetrics(metricsClient);
+      List<Metric> metrics = getMetrics(metricsClient);
 
       for (String metric : TAGGED_METRICS) {
-         expectNycSiteTag(true, metric, metricsAndTags);
+         expectNycSiteTag(true, metric, metrics);
       }
 
       for (String metric : UNTAGGED_METRICS) {
-         expectNycSiteTag(false, metric, metricsAndTags);
+         expectNycSiteTag(false, metric, metrics);
+      }
+   }
+
+   @Test
+   public void testJGroupsDetailedMetrics() {
+      RestMetricsClient metricsClient = SERVERS.rest(LON).get().metrics();
+
+
+      List<Metric> metrics = getMetrics(metricsClient);
+
+      // in 3 nodes cluster, we should have 3 metrics with the same name and different tags
+      // async requests counter
+      assertDetailedMetrics(metrics, "vendor_jgroups_stats_async_requests_total", false);
```

Comment:
Maybe I'm misunderstanding what's being provided, ut shouldn't we provide different metrics for internal cluster communication and XSite traffic so that the user can have a histogram of each? I'm just thinking that the expected latencies of the former will be much smaller than the latter.

Reply:
I'm not collecting cross-site metrics here because JGroups will pick a random node in the remote site. All these metrics are for cluster communication. I just put it here since I had the class in place to parse the output from REST :rofl:

Reply:
Argh I see :slightly_smiling_face:

Can we split this out into it's own test case to make things more explicit then?

Reply:
I changed `RestMetricsResource` to receive this test but, after changing, I realized the clustered tests only use 2 nodes cluster. The cross-site tests uses 3 nodes which allows me to test that I have duplicated metrics name with different tags. 
Checkout `RestMetricsResource` changes and let me know if I should revert them :+1:

Reply:
I think the changes are good. We could update `RestMetricsResource` further so that it uses it's own cluster, not `ClusteredIT.Servers` with 3 nodes. That way we don't have to have any code in the XSite tests.